src/lib/Settings.svelte

```diff
@@ -1,32 +1,122 @@
+        scannerPath = await settings.scannerPath ?? '';
+        donePath = await settings.donePath ?? '';
     })
 
-    function dispatchSave() {
-        dispatch("save", {
-            newScanPath: tmpScannerPath,
-            newDonePath: tmpDonePath
-        })
+    async function pickScannerPath() {
+        await invoke<string>('pick_directory', {startPath: scannerPath})
+            .then(path => saveScannerPath(path))
+            .catch(error => console.log(error))
+    }
+
+    async function pickDonePath() {
+        await invoke<string>('pick_directory', {startPath: donePath})
+            .then(path => saveDonePath(path))
+            .catch(error => console.log(error))
     }
-</script>
 
-<form on:submit|preventDefault={() => dispatchSave()}>
-    <label for="scannerPath">Skanner kilde</label>
-    <input type="text" id="scannerPath" bind:value={tmpScannerPath}/>
+    function saveScannerPath(path: string): void {
+        readDir(path)
+            .then(() => {
+                scanPathError = undefined
+                scannerPath = path
+                dispatch("save", {newScanPath: path})
+                scanPathSuccess = "Lagret!"
+                setTimeout(() => scanPathSuccess = undefined, 5000)
+            })
+            .catch((e: string) => {
+                if (e.includes('Not a directory')) scanPathError = "Dette er ikke en mappe"
+                else scanPathError = "Mappen eksisterer ikke"
+            })
+    }
 
-    <label for="donePath">Mappe for ferdige objekter</label>
-    <input type="text" id="donePath" bind:value={tmpDonePath}/>
+    function saveDonePath(path: string): void {
+        readDir(path)
+            .then(() => {
+                donePathError = undefined
+                donePath = path
+                dispatch("save", {newDonePath: path})
+                donePathSuccess = "Lagret!"
+                setTimeout(() => donePathSuccess = undefined, 5000)
+            })
+            .catch((e: string) => {
+                if (e.includes('Not a directory')) donePathError = "Dette er ikke en mappe"
+                else donePathError = "Mappen eksisterer ikke"
+            })
+    }
 
-    <button type="submit">Save</button>
+</script>
+
+<form on:submit|preventDefault class="settings-form">
+    <div class="form-group">
+        <label for="scannerPath">Skanner kilde</label>
+        <button on:click={() => pickScannerPath()}>Velg mappe</button>
+        <input type="text" id="scannerPath" bind:value={scannerPath}/>
+        <button on:click={() => saveScannerPath(scannerPath)}>Lagre</button>
+        {#if scanPathError}
+            <p class="error-message">{scanPathError}</p>
+        {/if}
+        {#if scanPathSuccess}
+            <p class="success-message">{scanPathSuccess}</p>
+        {/if}
+    </div>
+
+    <div class="form-group">
+        <label for="donePath">Ferdige objekter</label>
+        <button on:click={() => pickDonePath()}>Velg mappe</button>
```

Comment:
```suggestion
        <button type="button" on:click={() => pickDonePath()}>Velg mappe</button>
```
Har som standard 'type="submit"', som gjør hvis vi trykker "enter" i inputfeltet så åpnes filvelgeren. Vil heller det skulle gått til lagre knappen